tobac/utils/general.py

```diff
@@ -411,11 +423,17 @@ def get_spacings(field_in, grid_spacing=None, time_spacing=None):
     ) and (grid_spacing is None):
         x_coord = deepcopy(field_in.coord("projection_x_coordinate"))
         x_coord.convert_units("metre")
-        dx = np.diff(field_in.coord("projection_y_coordinate")[0:2].points)[0]
+        dx = np.diff(x_coord[0:2].points)[0]
         y_coord = deepcopy(field_in.coord("projection_y_coordinate"))
         y_coord.convert_units("metre")
-        dy = np.diff(field_in.coord("projection_y_coordinate")[0:2].points)[0]
-        dxy = 0.5 * (dx + dy)
+        dy = np.diff(y_coord[0:2].points)[0]
+
+        if average_method == "arithmetic":
+            dxy = 0.5 * (np.abs(dx) + np.abs(dy))
+        elif average_method == "geometric":
+            # I would prefer a geometric mean
+            dxy = np.sqrt(np.abs(dx) * np.abs(dy))
+
     elif grid_spacing is not None:
         dxy = grid_spacing
     else:
```

Comment:
should we raise a similar value error if time_spacing is None, but we cannot find the information from the coordinates?

Reply:
yes, well spotted! I added a corresponding values error:

```diff
         # use value of time_spacing for dt:
         dt = time_spacing
+    else:
+        raise ValueError(
+            "no information about time spacing, need either input cube with time or keyword argument time_spacing"
+        )
     return dxy, dt
```

Reply:
We should change the if statement to: `if ("time" in coord_names) and (time_spacing is None)` as well then to check whether a time coordinate is present

Reply:
> We should change the if statement to: `if ("time" in coord_names) and (time_spacing is None)` as well then to check whether a time coordinate is present

@JuliaKukulies : Sorry, I missed this comment and already merged back. Could you open another issue on this suggested change?